core/codegen/src/ident.rs

```diff
@@ -0,0 +1,99 @@
+//! A variant of Ident with more ergonomic handling of raw identifiers.
+
+use std::cmp::Ordering;
+
+use devise::syn;
+use self::syn::ext::IdentExt as _;
+use crate::proc_macro2::{Ident as Ident2, Span};
+use quote::ToTokens;
+
+#[repr(transparent)]
+#[derive(Clone, Debug, Eq, Ord)]
+pub struct Ident(Ident2);
+
+impl Ident {
+    pub fn is_raw(&self) -> bool {
+        self.0.to_string().starts_with("r#")
+    }
+
+    pub fn name(&self) -> String {
+        self.0.unraw().to_string()
+    }
+
+    pub fn prepend(&self, string: &str) -> Ident {
+        // TODO: want to use r#{}{}, but that fails (`"r#__rocket_param_enum"` is not a valid identifier)
```

Comment:
`proc_macro2::Ident` has a separate `new_raw` constructor for this, but it's super-extra-unstable.